climada/util/calibrate/base.py

```diff
@@ -158,6 +159,102 @@ def plot_impf_set(self, **plot_kwargs):
         """
         return self.impf_set.plot(**plot_kwargs)
 
+    def plot_impf_variability(
+        self,
+        cost_func_diff: float = 0.1,
+        p_space_df: Optional[pd.DataFrame] = None,
+        plot_haz: bool = True,
+        plot_impf_kws : dict = {},
+        plot_hist_kws : dict = {},
+    ):
+
+        """Plot impact function variability with parameter combinations of
+        almost equal cost function values
+
+        Args:
+            cost_func_diff (float, optional): Max deviation from optimal cost
+            function value (as fraction). Defaults to 0.1 (i.e. 10%).
+            p_space_df (pd.DataFrame, optional): parameter space. Defaults to None.
+            plot_haz (bool, optional): Whether or not to plot hazard intensity
+            distibution. Defaults to False.
+        """
+
+        haz_type = self.input.hazard.haz_type
+        if p_space_df is None:
+            p_space_df = self.output.p_space_to_dataframe()
```

Comment:
Big problem: The `OutputEvaluator` takes any `Output` object, but `p_space_to_dataframe()` is only available from `BayesianOptimizerOutput`. To avoid confusion, it might make sense the derive a `BayesianOutputEvaluator` from `OutputEvaluator` and move this function there, or have a type check in the beginng of the function. Sorry I didn't notice that before 🙈

Reply:
Good point. In my opinion a type check at the beginning of the function would be a good solution.

Reply:
Alright, let's do it like this for now.